### ISSUE TITLE
question about the code 'centers_logits = F.normalize(self.head_fc(self.fc.weight.T), dim=1)'

### ISSUE THREAD

**Issue by byte-dance** [FlamieZhu/BCL#9]
class BCLModel(nn.Module):
   
        self.fc = nn.Linear(dim_in, num_classes)
        self.head_fc = nn.Sequential(nn.Linear(dim_in, dim_in), nn.BatchNorm1d(dim_in), nn.ReLU(inplace=True),
                                   nn.Linear(dim_in, feat_dim))
    def forward(self, x):
        centers_logits = F.normalize(self.head_fc(self.fc.weight.T), dim=1)

     self.fc.weight.T has shape (dim_in , num_classes), how can it be sent to self.head_fc ?

**Comment by adv010** [comment]
@byte-dance I encountered the same issue. For now Ive changed self.head_fc to be : 
`        self.head_fc = nn.Sequential(nn.Linear(num_classes, dim_in), nn.BatchNorm1d(dim_in), nn.ReLU(inplace=True),
                                   nn.Linear(dim_in, feat_dim))`

**Comment by lgX1123** [comment]
> Specifically, we have class-specific weights w1, w2, . . . , wK after a nonlinear transformation MLP as prototypes zc1 , zc2 , . . . , zcK .

Maybe authors want to get centers_logits in the shape of (class, feat_dim). So, I think remove `.T` would be better?

**Comment by ericyq** [comment]
code is right. running successfully

**Comment by ericyq** [comment]
code is right. running successfully!

features = torch.cat([feat.unsqueeze(1), feat.unsqueeze(1)], dim=1) # (N,2,dim)
scl_loss = criterion_scl(centers, features, targets) # centers: [class_num, dim]

**Comment by ericyq** [comment]
code is right. running successfully!

features = torch.cat([feat.unsqueeze(1), feat.unsqueeze(1)], dim=1) # (N,2,dim)
scl_loss = criterion_scl(centers, features, targets) # centers: [class_num, dim]
